libraries/pbrlib/genglsl/mx_sheen_brdf.glsl

```diff
@@ -1,13 +1,52 @@
 #include "pbrlib/genglsl/lib/mx_bsdfs.glsl"
 
-void mx_sheen_brdf_reflection(vec3 L, vec3 V, float weight, vec3 color, float roughness, vec3 normal, BSDF base, out BSDF result)
+void mx_sheen_brdf_reflection(vec3 L, vec3 V, float weight, vec3 color, float roughness, vec3 N, BSDF base, out BSDF result)
 {
-    // TODO: implement this
+    if (weight < M_FLOAT_EPS)
+    {
+        result = base;
+        return;
+    }
+
+    float NdotL = dot(N,L);
+    float NdotV = dot(N,V);
+    if (NdotL <= 0.0 || NdotV <= 0.0)
+    {
+        result = base;
+        return;
+    }
+
+    vec3 H = normalize(L + V);
+    float NdotH = dot(N, H);
+
+    float alpha = clamp(roughness, M_FLOAT_EPS, 1.0);
+    float D = mx_microfacet_sheen_NDF(NdotH, alpha);
+
+    vec3 F = color * weight;
+
+    // Geometry term is skipped and we use a smoother denominator, as in:
+    // https://blog.selfshadow.com/publications/s2013-shading-course/rad/s2013_pbs_rad_notes.pdf
+    vec3 fr = D * F / (4.0 * (NdotL + NdotV - NdotL*NdotV));
```

Comment:
Divide by 0 test in case the vectors are all aligned?

Reply:
Both dot products are check for being zero or negative above. Since they both are numbers in range (0.0, 1.0] the denominator cannot become zero here. (the sum of adding them will always be greater than the product of multiplying them)